core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -444,11 +444,20 @@ private[spark] object JsonProtocol {
     ("Disk Size" -> rddInfo.diskSize)
   }
 
-  def storageLevelToJson(storageLevel: StorageLevel): JValue = {
-    ("Use Disk" -> storageLevel.useDisk) ~
-    ("Use Memory" -> storageLevel.useMemory) ~
-    ("Deserialized" -> storageLevel.deserialized) ~
-    ("Replication" -> storageLevel.replication)
+  def storageLevelToJson(storageLevel: StorageLevel): JValue = storageLevel match {
+    case StorageLevel.NONE => "NONE"
```

Comment:
Maybe we can change `StorageLevel.toString` to do this or add another method e.g. `StorageLevel.name`.

Reply:
Agree, that would be more robust.